backend/routes/userRoutes.js

```diff
@@ -27,15 +47,59 @@ router.get("/getteams",protect,getTeamById);
 
 router.post("/login", login);
 router
-  .route("/profile")
-  .get(protect, getUserProfile)
-  .put(protect, updateUserProfile);
+  .route("/profile/")
+  .get(getUserProfile);
```

Comment:
May be remove getUserProfile and replace it with getUserProfileByUsername function. Dont change the route name. 
And Remove this function
```
router.get(
  "/getuserprofile",
  checkQuery("username").not().isEmpty().trim().escape(),
  getUserProfileByUsername
);
```
 if its not there do a rebase with main. you can refer https://github.com/veerreshr/team-maker/blob/main/CONTRIBUTING-gitflow.md for rebasing